driver-core/src/main/java/com/datastax/driver/core/Responses.java

```diff
@@ -448,6 +449,8 @@ private Rows(Metadata metadata, Queue<List<ByteBuffer>> data, ProtocolVersion ve
             public String toString() {
                 StringBuilder sb = new StringBuilder();
                 sb.append("ROWS ").append(metadata).append('\n');
+                // using default codec registry as this is only intended for debugging purposes
+                CodecRegistry codecRegistry = new CodecRegistry();
```

Comment:
We could just have a 'default registry' that is package protected so it can't be accessed/mutated from the outside.  We would just have to be careful not to mutate it from the inside :)